lib/connection.ts

```diff
@@ -533,14 +533,27 @@ export class Connection extends Entity {
    */
   createSession(options?: SessionCreateOptions): Promise<Session> {
     return new Promise((resolve, reject) => {
+      const abortSignal = options && options.abortSignal;
+      let rejectOnAbort: Func<void, void>;
+      if (abortSignal) {
+        rejectOnAbort = () => {
+          const err = createAbortError();
+          log.error("[%s] [%s]", this.id, err.message);
+          return reject(err);
+        };
+        if (abortSignal.aborted) {
```

Comment:
The early exit check is made towards the end just before deciding whether or not to add event listener to the abort signal. Now that we are doing the same here, we can remove that check